### PR TITLE
Implements $cond operator in aggregate expressions.

### PR DESCRIPTION
**Pull request by vladlosev**
@parkr @Phraxos @bigthyme This adds the `$cond` operator which can provide default values in aggregate expressions.

### CODE REVIEW

File: lib/aggregate/expression.js [hunk 1]
```diff
@@ -79,15 +79,66 @@ function validateExpression(expression) {
           }
           break;
         case '$size':
-         if (_.isArray(params) && params.length !== 1) {
+          if (_.isArray(params) && params.length !== 1) {
+             return aggregateUtils.getErrorReply(
+               util.format(
+                 'exception: Expression $size takes exactly 1 arguments. ' +
```
Comment (parkr) [inline lib/aggregate/expression.js]: exactly 1 argument? why pluralize?

**Comment by parkr** [comment]
Looks great!!!!

File: lib/aggregate/expression.js (hunk 1 above)
Comment (vladlosev) [inline lib/aggregate/expression.js]: Going along with the source:
```
16:05:19 local-vlad-Vlads-MacBook-Air vlad mongodb-fs  > mongo
MongoDB shell version: 2.6.8
connecting to: test
> db.test.aggregate([{$group: {_id: '$a', total: {$sum: {$size: ['$b', '$c']}}}}])
assert: command failed: {
	"errmsg" : "exception: Expression $size takes exactly 1 arguments. 2 were passed in.",
	"code" : 16020,
	"ok" : 0
} : aggregate failed
Error: command failed: {
	"errmsg" : "exception: Expression $size takes exactly 1 arguments. 2 were passed in.",
	"code" : 16020,
	"ok" : 0
} : aggregate failed
    at Error (<anonymous>)
    at doassert (src/mongo/shell/assert.js:11:14)
    at Function.assert.commandWorked (src/mongo/shell/assert.js:244:5)
    at DBCollection.aggregate (src/mongo/shell/collection.js:1149:12)
    at (shell):1:9
2015-05-28T16:05:22.774+0000 Error: command failed: {
	"errmsg" : "exception: Expression $size takes exactly 1 arguments. 2 were passed in.",
	"code" : 16020,
	"ok" : 0
} : aggregate failed at src/mongo/shell/assert.js:13
>
```

**Comment by bigthyme** [comment]
:shipit:

**Comment by vladlosev** [comment]
Thanks for the review guys!

**Pull request merged**